lib/providers/credential.js

```diff
@@ -1,22 +1,28 @@
 /* global Config */
 'use strict';
 
+const Interface = require('contracts-es6');
+const ProviderInterface = require('./interface');
+
 /**
  * Provider for Credential secrets
  */
-class CredentialProvider {
+class CredentialProvider extends Interface.UnboundedInterface(ProviderInterface) {
     /**
    * Create a new instance of the `GenericProvider`.
    *
    * @param {string} path
    * @param {string} token
    */
   constructor(path, token) {
+    super();
+
     const splitPath = path.split('/');
 
     this.mount = splitPath[0];
     this.role = splitPath[1];
     this.token = token;
+    this.data = null;
```

Comment:
Is the `CredentialProvider` doing anything with `this.data`? Or should we just leave it out an have an empty `invalidate()` function?